yeat/tests/test_paired.py

```diff
@@ -7,83 +7,87 @@
+import json
 import pandas as pd
 from pathlib import Path
 import pytest
 import re
 import subprocess
-from yeat import cli, workflows
+from yeat import cli
 from yeat.tests import data_file
 
 
-def test_paired_end_read_assemblers_dry_run(tmp_path):
+def test_paired_end_assemblers_dry_run(tmp_path):
     """The purpose of this test is to check if the snakemake workflow is properly defined."""
     wd = str(tmp_path)
     arglist = [
         "--outdir",
         wd,
-        "--sample",
-        "Animal_289",
         "-n",
-        "--paired",
-        data_file("Animal_289_R1.fq.gz"),
-        data_file("Animal_289_R2.fq.gz"),
-        data_file("config.cfg"),
+        data_file("configs/paired.cfg"),
     ]
     args = cli.get_parser().parse_args(arglist)
     cli.main(args)
 
 
-def test_invalid_read1_file():
-    with pytest.raises(Exception, match=r"No such file:.*read1\'"):
-        read1 = "read1"
-        read2 = "read2"
-        assemblers = ["spades"]
-        workflows.run_paired(read1, read2, assemblers)
-
-
-def test_invalid_read2_file():
-    with pytest.raises(Exception, match=r"No such file:.*read2\'"):
-        read1 = data_file("short_reads_1.fastq.gz")
-        read2 = "read2"
-        assemblers = ["spades"]
-        workflows.run_paired(read1, read2, assemblers)
+@pytest.mark.long
+@pytest.mark.parametrize(
+    "algorithm,label,expected",
+    [
+        ("spades", "spades-meta", "contigs.fasta"),
+        ("megahit", "megahit-default", "final.contigs.fa"),
+        ("unicycler", "unicycler-default", "assembly.fasta"),
+    ],
+)
+def test_individual_paired_end_assemblers(algorithm, label, expected, capsys, tmp_path):
+    wd = str(tmp_path)
+    arglist = [
+        "--outdir",
+        wd,
+        data_file(f"configs/{algorithm}.cfg"),
+    ]
+    args = cli.get_parser().parse_args(arglist)
+    cli.main(args)
+    observed = Path(wd).resolve() / "analysis" / "sample1" / label / algorithm / expected
+    assert observed.exists()
 
 
 @pytest.mark.long
-def test_multiple_assemblers(capsys, tmp_path):
+def test_multiple_spades(capsys, tmp_path):
     wd = str(tmp_path)
     arglist = [
         "--outdir",
         wd,
-        "--paired",
-        data_file("short_reads_1.fastq.gz"),
-        data_file("short_reads_2.fastq.gz"),
-        data_file("config.cfg"),
+        data_file(f"configs/two_spades.cfg"),
     ]
     args = cli.get_parser().parse_args(arglist)
     cli.main(args)
-    spades_result = Path(wd).resolve() / "analysis" / "spades" / "contigs.fasta"
-    assert spades_result.exists()
-    megahit_result = Path(wd).resolve() / "analysis" / "megahit" / "final.contigs.fa"
-    assert megahit_result.exists()
+    analysis_dir = Path(wd).resolve() / "analysis"
+    expected = [
+        analysis_dir / "sample1" / "spades-default" / "spades" / "contigs.fasta",
+        analysis_dir / "sample1" / "spades-meta" / "spades" / "contigs.fasta",
+    ]
+    for observed in expected:
+        assert observed.exists()
```

Comment:
I use "observed" and "expected" as variable names frequently in test functions. I'm happy to see you picking that up, but I'm not sure the names are used appropriately in a couple cases here. Usually I use observed and expected to refer to the content or value of a variable.

```python
observed = some_function(test_data)
expected = 42
assert observed == expected
```

Here, you're referring to paths that you expect to exist. So the line `for observed in expected` is really misleading according to that pattern. More appropriate would be:

```python
expected_paths = [
    analysis_dir / "sample1" / "spades-default" / "spades" / "contigs.fasta",
    analysis_dir / "sample1" / "spades-meta" / "spades" / "contigs.fasta",
]
for expected in expected_paths:
    assert expected_paths.exists()
```

And honestly, this would be even more clear and concise.

```
assert (analysis_dir / "sample1" / "spades-default" / "spades" / "contigs.fasta").exists()
assert (analysis_dir / "sample1" / "spades-meta" / "spades" / "contigs.fasta").exists()
```